### DISCUSSION TITLE
Build failing when adding in YamlDotNet package

### DISCUSSION THREAD

**Discussion by devzaak** [cake-build/cake#3415] [Extension Q&A]
Hi, 
I am using cake for builds, and all works fine until I try and add:
`#addin nuget:?package=YamlDotNet&version=11.2.1`

This results in script compilation failing with:
```
Resolving assembly Microsoft.CodeAnalysis.CSharp.resources, Version=3.9.0.0, Culture=en-GB, PublicKeyToken=31bf3856ad364e35
Exception while resolving assembly Microsoft.CodeAnalysis.CSharp.resources: Could not load file or assembly 'Microsoft.CodeAnalysis.CSharp.resources, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.
```
Naturally I tried adding in the reference for `Microsoft.CodeAnalysis.CSharp` via:
`#addin nuget:?package=Microsoft.CodeAnalysis.CSharp&version=3.9.0`
but that does not seem to have any effect.

Any ideas on what I might be doing wrong?

**Comment by augustoproiete** [comment]
@aslezak What's the Cake version you're using, [runner type](https://cakebuild.net/docs/running-builds/runners/) (e.g. .NET Tool, Frosting, etc.), and operating system where you're running Cake?

I tried to repro the above using the latest Cake version as of this writing (1.1.0), [.NET Tool](https://cakebuild.net/docs/running-builds/runners/dotnet-tool) runner, on Windows 10, and worked as expected:

```cakescript
#addin "nuget:?package=YamlDotNet&version=11.2.1"

using YamlDotNet.Serialization;
using YamlDotNet.Serialization.NamingConventions;

var person = new
{
    Name = "Abe Lincoln",
    Age = 25,
};

var serializer = new SerializerBuilder()
    .WithNamingConvention(CamelCaseNamingConvention.Instance)
    .Build();

var yaml = serializer.Serialize(person);
Information(yaml);
```

![image](https://user-images.githubusercontent.com/177608/127586560-5872565a-5469-4684-bef9-a70eabd90f02.png)

**Reply by devzaak** [reply]
Ok, that is very strange, as I am using the exact same settup, dotnet tool with cake v1.1.0
I wonder if the problem is coming from some combination of packages I am loading.

this is the full `bootstrap.cake` of my project:

```
#addin nuget:?package=mustache-sharp&version=1.0.0

// Install YamlDotNet as a Cake Addin
#addin nuget:?package=YamlDotNet&version=11.2.1

// explicitly add transitive dependencies
#addin nuget:?package=AWSSDK.Core&version=3.5.3.1
#addin nuget:?package=AWSSDK.S3&version=3.5.8.7
#addin nuget:?package=AWSSDK.ECR&version=3.5.1.32
#addin nuget:?package=DotNet.Xdt&version=2.2.1
#addin nuget:?package=MimeTypesMap&version=1.0.8

#addin nuget:?package=Cake.AWS.S3&version=1.0.0
#addin nuget:?package=Cake.Docker&version=1.0.0
#addin nuget:?package=Cake.FileHelpers&version=4.0.1
#addin nuget:?package=Cake.Incubator&version=6.0.0
#addin nuget:?package=Cake.XdtTransform&version=1.0.0

#tool dotnet:?package=dotnet-reportgenerator-globaltool&version=4.8.8
#tool dotnet:?package=GitVersion.Tool&version=5.6.9



var Build = new Builder(BuildSystem, Context, target => RunTarget(target));
```

**Reply by augustoproiete** [reply]
Strange indeed. I ran the same test above with all these dependencies, and it worked normally without issues.

Maybe it's related to the .NET Core SDK you have installed on the machine or the SDK that is being forced via a `global.json` on the directory you're running the Cake .NET Tool? :thinking:

This is what I see on `dotnet --info` on my test VM for Cake:

```
.NET SDK (reflecting any global.json):
 Version:   5.0.302
 Commit:    c005824e35

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.302\
```

**Reply by devzaak** [reply]
more or less the same here with additional SDKs installed
```
.NET SDK (reflecting any global.json):
 Version:   5.0.302
 Commit:    c005824e35

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19043
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.302\

Host (useful for support):
  Version: 5.0.8
  Commit:  35964c9215

.NET SDKs installed:
  2.2.207 [C:\Program Files\dotnet\sdk]
  3.1.411 [C:\Program Files\dotnet\sdk]
  5.0.205 [C:\Program Files\dotnet\sdk]
  5.0.302 [C:\Program Files\dotnet\sdk]
  ```
  
  I wonder if I managed to get this whole system in funky state, I will try building from YamlDotNet and going the other way in clean project. Getting the feeling it is more about my faulty testing process than issue with libraries at this point.
  Thanks for help!
